harness/determined/common/experimental/experiment.py

```diff
@@ -259,10 +259,19 @@ def get_with_offset(offset: int) -> bindings.v1GetExperimentCheckpointsResponse:
             sort_by = sb
             smaller_is_better = config.get("searcher", {}).get("smaller_is_better", True)
 
+        reverse = not smaller_is_better
+
         def key(ckpt: checkpoint.Checkpoint) -> Any:
             training = ckpt.training
             assert training
-            return training.validation_metrics["avgMetrics"][sort_by]
+            metric = training.validation_metrics.get("avgMetrics") or {}
+            metric = metric.get(sort_by)
+
+            # Return a bool here to sort checkpoints that may have no validation metrics.
+            if reverse:
+                return metric is not None, metric
+            else:
+                return metric is None, metric
```

Comment:
what about just: `return (metric is None) == reverse, metric`?

Or possibly the `!= reverse`, I can't seem to think through it in my head.

Maybe that's a reason to leave it as written though.

Reply:
yeah, i went through this logic loop too to try and simplify the bools. also couldn't seem to easily think through it lol. i think it's less convoluted to read this way.